pandas/tests/dtypes/test_inference.py

```diff
@@ -1200,6 +1200,20 @@ def test_interval(self):
         inferred = lib.infer_dtype(pd.Series(idx), skipna=False)
         assert inferred == "interval"
 
+    def test_string_dtype(self):
+        # StringArray
```

Comment:
can you parametrize these according to the klass, also add a numpy object array to this list
and test for both an array with a pd.NA and w/o

Reply:
>  also add a numpy object array to this list

numpy object is already tested elsewhere, this is specifically for string dtype (so, yeah, the list doesn't really fit then, I can also remove that).